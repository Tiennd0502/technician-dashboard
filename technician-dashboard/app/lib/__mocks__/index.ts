export * from './table';
export * from './categories';
export * from './services';
