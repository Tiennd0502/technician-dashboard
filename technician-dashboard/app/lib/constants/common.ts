export const LOGO_URL = '/images/technician.svg';

export const AVATAR_DEFAULT = '/images/empty-avatar.svg';

export const CURRENCY_CHARACTERS = {
  EURO: 'â‚¬',
};

export const TRUNCATE_STYLE = {
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
};

export const NO_DATA = 'No data found';
