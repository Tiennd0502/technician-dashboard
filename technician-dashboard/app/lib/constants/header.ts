import { AVATAR_DEFAULT } from '.';

export const HEADER_INFO = {
  user: {
    name: 'Belle',
    avatar: AVATAR_DEFAULT,
  },
  title: 'Product & Services',
};
