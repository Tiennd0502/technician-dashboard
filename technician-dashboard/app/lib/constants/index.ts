export * from './common';
export * from './filter';
export * from './sidebar';
export * from './header';
export * from './rules';
export * from './messages';
export * from './routes';
export * from './status';
export * from './table';
export * from './categories';
export * from './services';
