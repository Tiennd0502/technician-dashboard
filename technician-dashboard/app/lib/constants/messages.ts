export const ERROR_MESSAGE = {
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required.`,
};
