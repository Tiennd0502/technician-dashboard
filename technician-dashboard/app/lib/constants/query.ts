export const QUERY_KEYS = {
  CATEGORIES: ['categories'],
  SERVICES: ['services'],
};
