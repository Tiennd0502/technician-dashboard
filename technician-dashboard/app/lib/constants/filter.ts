import { SORT_TYPE } from '@/lib/interfaces';

export const PRODUCT_LIMIT = 5;

export const DEFAULT_PRODUCT_FILTER = {
  name: '',
  sortBy: 'name',
  order: SORT_TYPE.Asc,
};
