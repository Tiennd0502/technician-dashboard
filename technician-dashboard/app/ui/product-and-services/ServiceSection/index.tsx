'use client';

import React, { memo } from 'react';
import { Flex, FlexProps, Heading, LayoutProps } from '@chakra-ui/react';
import { useQuery } from '@tanstack/react-query';

// Types
import { Category, Service } from '@/lib/interfaces';

// Constants
import { API_ROUTES, QUERY_KEYS } from '@/lib/constants';

// Components
import { Spinner } from '@/ui/commons';
import { Categories, ServiceList } from '@/ui/product-and-services';

// Utils
import { getCategories, getServices } from '@/lib/utils/services';

interface ServiceSectionProps extends FlexProps {
  width: LayoutProps['w'];
}

const ServiceSection = ({ width, ...props }: ServiceSectionProps) => {
  const { data: categories = [] } = useQuery({
    queryKey: QUERY_KEYS.CATEGORIES,
    queryFn: () => getCategories(),
  });

  const { data: services = [] } = useQuery({
    queryKey: QUERY_KEYS.SERVICES,
    queryFn: () => getServices(),
  });

  return (
    <Flex
      w={width}
      flexDirection='column'
      justifyContent='space-between'
      gap='5'
      borderWidth='1px'
      borderColor='border.primary'
      borderRadius='md'
      p='7'
      minH='315px'
      order={{
        md: 1,
        xl: 1,
      }}
      {...props}
    >
      <Heading variant={{ base: 'headingLg', lg: 'headingXl' }}>About Services</Heading>
      <ServiceList list={services as Service[]} />
      <Categories list={categories as Category[]} />
    </Flex>
  );
};

export default memo(ServiceSection);
