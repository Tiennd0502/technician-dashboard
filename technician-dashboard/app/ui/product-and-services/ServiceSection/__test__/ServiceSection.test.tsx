import { render, renderHook, waitFor } from '@testing-library/react';

import * as ReactQuery from '@tanstack/react-query';
import { ChakraProvider } from '@chakra-ui/react';

import { API_ROUTES, CATEGORIES, SERVICES } from '@/lib/constants';

import ServiceSection from '..';
import { useFetchServices } from '@/lib/hooks';

const queryClient = new ReactQuery.QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

jest.mock('@/lib/hooks', () => ({
  ...jest.requireActual('@/lib/hooks'),
  useFetchServices: ([]) => [
    {
      data: [],
      isLoading: false,
    },
    {
      data: [],
      isLoading: false,
    },
  ],
}));

const props = {
  width: '100%',
};

const setup = () =>
  render(
    <ChakraProvider>
      <ReactQuery.QueryClientProvider client={queryClient}>
        <ServiceSection {...props} />
      </ReactQuery.QueryClientProvider>
    </ChakraProvider>,
  );

describe('ServiceSection test cases', () => {
  test('should render correctly', async () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
