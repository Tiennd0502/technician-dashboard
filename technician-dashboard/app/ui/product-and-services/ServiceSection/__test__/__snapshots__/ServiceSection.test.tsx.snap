// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceSection test cases should render correctly 1`] = `
<div>
  <div
    class="css-pbxem"
  >
    <h2
      class="chakra-heading css-1dklj6k"
    >
      About Services
    </h2>
    <div
      class="css-166re2f"
      data-testid="list-services"
    />
    <div
      class="css-166re2f"
      data-testid="list-category"
    />
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
